### DISCUSSION TITLE
pdns-server fails to start - can't find an exiting file

### DISCUSSION THREAD

**Discussion by Brownowl** [PowerDNS/pdns#13643] [Q&A] [answered]
Hi folks,

I suspect this plea for help is more appropriate for the mailing list, but I've sent two emails in the last few days and they seem to have fallen into a black hole. And yes, I still have a valid account and email on the list.

Anyway, I hope this isn't too far out of bounds, and I would really appreciate some help with this please. It's probably something simple, but I just can't see it however hard I look.

I've been using pdns happily with mysql and a tinydns backend for several years now on a couple of Ubuntu machines. These now need to be replaced, so I've installed all I need on a new one. However, I have a problem getting pdns to run, and I'm really struggling to work out what's causing the problem.

I get the following messages on startup:

```
pdns_server[262519]: Failed to open cdb database file
'/var/tmp/data.cdb': No such file or directory

pdns_server[262519]: Fatal error: Failed to open cdb database file
'/var/tmp/data.cdb': No such file or directory

systemd[1]: pdns.service: Main process exited, code=exited, status=1/FAILURE

systemd[1]: pdns.service: Failed with result 'exit-code'.
systemd[1]: Failed to start PowerDNS Authoritative Server.
```

However, the file concerned is very much there!

```
# ls -al /var/tmp/data.cdb
-rw-r--r-- 1 root root 56678 Dec 15 10:55 /var/tmp/data.cdb

ls -al /var
drwxrwxrwt 11 root   root   4096 Dec 15 12:49 tmp
```

I've tried `chmod 777` and `chown pdns:pdns` to no avail. I've set logging
to debug, but there are no clues there either. It has exactly the same
permissions and ownership as on the old machines.

The setup script I followed is the one I wrote when I installed it on
the older machines, and I've checked and re-checked the config files,
which as far as I can see are identical, but I'm stumped.

I'm sure I'm missing something simple, but however hard I look and use
Google I can't see it. So I'd really appreciate some help please.

New platform is Ubuntu Server 22.04.3 LTS and pdns-server version is
4.8.3 including tinydns and mysql backends).

In case it's relevant, the old machines are Ubuntu Server 20.04.6 LTS
and pdns version 4.2.1-1build2

Many thanks, Laurie.

**Comment by rgacogne** [comment, marked as answer]
I'm pretty sure this is because we set `PrivateTmp` in our systemd unit file, for security reasons: https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#PrivateTmp=

That means PowerDNS doesn't see the global `/tmp` and `/var/tmp/` directories but their own private ones. I would suggest moving your `cdb` file to a different location. If you cannot do that, you can set `PrivateTmp=false` via `systemctl edit --full pdns`.

**Reply by Brownowl** [reply]
Many, many thanks. That was exactly it! Completely news to me, and I can't imagine I would ever have found that on my own!

Thanks again - this can be marked as solved. Yay!

Cheers, Laurie.

PS. Still no success getting an email into the mailing list.

**Reply by Habbie** [reply]
> PS. Still no success getting an email into the mailing list.

I see why. I just sent you an email :)
